uitools/toolkitcpp/import/Esri/ArcGISRuntime/Toolkit/AttachmentsPopupElementView.qml

```diff
@@ -132,25 +128,39 @@ ColumnLayout {
                         model.listModelData.downloadAttachment();
                     } else {
                         if (model.listModelData.popupAttachmentType === QmlEnums.PopupAttachmentTypeImage) {
-                            fullScreenImage.source = model.listModelData.localData;
-                            fullScreenImageDialog.visible = true;
+                            if (popupView.openImagesInternally) {
+                                fullScreenImage.source = model.listModelData.localData;
+                                fullScreenImageDialog.visible = true;
+                            } else {
+                                // user disabled default behavior, so we do nothing
+                            }
                         } else {
-                            // This is a known limitation iOS but there are a few mentioned workarounds, https://bugreports.qt.io/browse/QTBUG-42942
-                            // Android currnetly has a bug logged against this, https://bugreports.qt.io/browse/QTBUG-133702
-                            // We plan on supporting these platforms in the future when it's implemented on iOS and fixed on Android.
-                            if (Qt.platform.os !== "android" && Qt.platform.os !== "ios") {
-                                Qt.openUrlExternally(model.listModelData.localData);
+                            if (popupView.openAttachmentsExternally) {
+                                // This is a known limitation iOS but there are a few mentioned workarounds, https://bugreports.qt.io/browse/QTBUG-42942
+                                // Android currnetly has a bug logged against this, https://bugreports.qt.io/browse/QTBUG-133702
+                                // We plan on supporting these platforms in the future when it's implemented on iOS and fixed on Android.
+                                if (Qt.platform.os !== "android" && Qt.platform.os !== "ios") {
+                                    Qt.openUrlExternally(model.listModelData.localData);
+                                }
+                            } else {
+                                // user disabled default behavior, so we do nothing
```

Comment:
does the clicked event for both cases? I don't see where that happens

Reply:
The signal is emitted from the `fetchDataAsync` call in PopupAttachmentItem, https://github.com/Esri/arcgis-maps-sdk-toolkit-qt/blob/b270df915e5cb408ee47cb5e1ce5a5faf5a6bbcb/uitools/common/src/PopupAttachmentItem.cpp#L147

i just added a comment there for clarification as i did it everywhere else.